src/webchat-ui/components/plugins/input/file/FileAttachmentSection.tsx

```diff
@@ -0,0 +1,174 @@
+    onRemoveFileFromList: (index: number) => void;
+    onClose: () => void;
+}
+
+interface IFileAttachmentSectionState {
+    isDropZoneVisible: boolean;
+}
+
+class FileAttachmentSection extends React.PureComponent<React.HTMLProps<HTMLDivElement> & IFileAttachmentSectionProps, IFileAttachmentSectionState> {    
+    fileInputRef: React.RefObject<HTMLInputElement>;
+    closeButtonRef: React.RefObject<HTMLButtonElement>;
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            isDropZoneVisible: false,
+        };
+        this.fileInputRef = React.createRef();
+        this.closeButtonRef = React.createRef();
+    }
+
+    handleSelectFile = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const {onAddFilesToList} = this.props;
+        const files = event.target.files;
+        const newFilesArray = Array.prototype.slice.call(files);
+        onAddFilesToList(newFilesArray);
+    }
+
+    handleUploadFile = (event) => {
+        event.preventDefault();
+        this.fileInputRef.current?.click();
+    }
+
+    handleDragEnter = (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+        
+        this.setState({ isDropZoneVisible: true });
+    };
+
+    handleDropZoneVisibility = (isDropZoneVisible: boolean) => {
+        this.setState({ isDropZoneVisible })
+    }
+
+    render() {
+        const { fileList, onClose, onAddFilesToList, onRemoveFileFromList } = this.props;
+        const { isDropZoneVisible } = this.state;
+        const isFileListEmpty = fileList?.length === 0;
+
+        return (
+            <>
+                <Wrapper onDragEnter={this.handleDragEnter} className={isDropZoneVisible ? "active-drop-zone" : ""}>
+                    <CloseButton
+                        onClick={onClose}
+                        aria-label="Close File Attacment"
+                        ref={this.closeButtonRef}
+                    >
+                        <CloseIcon />
+                    </CloseButton>
+                    <FileUploadContainer>
+                        <AttachFileIcon />
+                        <DragDropTypography
+                            className="webchat-input-drag-and-drop-file-text"
+                        >
+                            Drop files or
+                        </DragDropTypography>
+                        <HiddenFileInput
+                            ref={this.fileInputRef}
+                            type="file"
+                            multiple
+                            onChange={this.handleSelectFile}
+                        />
+                        <AttachFileButton
+                            className="webchat-input-button-browse-files"
+                            aria-label="Browse Files"
+                            onClick={this.handleUploadFile}
+                        >
+                            Browse
+                        </AttachFileButton>
+                        {isFileListEmpty ? (
```

Comment:
this looks a bit hacky, why not work with a height on UploadedFilesContainer in PreviewUploadedFiles?